### PR TITLE
SGX documentation

### PR DESCRIPTION
**Pull request by mythi**
Closes: #456

### CODE REVIEW

**Comment by mythi** [comment]
/cc @ipuustin 

@intelkevinputnam this is a docs-only PR. I could use your feedback too, thanks!

File: cmd/sgx_plugin/README.md [hunk 1]
```diff
@@ -142,4 +205,69 @@ I0626 20:33:01.414446  964346 server.go:219] Start server for provision at: /var
...
+
+##### Build the image
+
+The demo uses container images build from Intel® SGX SDK and DCAP releases.
+
+To build the demo images:
+
+```bash
+$ cd ${INTEL_DEVICE_PLUGINS_SRC}
+$ make sgx-aesmd-demo
```
Comment (dcmiddle) [inline cmd/sgx_plugin/README.md:234]: I don't see these make targets. Does this depend on another PR?

File: README.md [hunk 2]
```diff
@@ -114,6 +115,48 @@ the card has:
...
+- the [admissing webhook](#admission-webhook)
+- the [SGX EPC memory registration](#sgx-epc-memory-registration)
+
+Brief overviews of the sub-components are given below.
+
+<!--
+The demo subdirectory contains a [video](demo#intel-sgx-device-plugin-demo-video) showing deployment
+and use of the SGX plugin.
+-->
+#### device plugin
```
Comment (intelkevinputnam) [inline README.md:139]: Please capitalize "Device"

File: README.md [hunk 3]
```diff
@@ -114,6 +115,48 @@ the card has:
...
+
+The [SGX device plugin](cmd/sgx_plugin/README.md) is responsible for discovering and reporting SGX
+device nodes to `kubelet`.
+
+#### admission webhook
```
Comment (intelkevinputnam) [inline README.md:144]: Please capitalize "Admission"

File: README.md [hunk 4]
```diff
@@ -114,6 +115,48 @@ the card has:
...
+
+The SGX admission webhook is responsible for performing Pod mutations based on what SGX features
```
Comment (intelkevinputnam) [inline README.md:146]: "which" SGX features.

File: cmd/sgx_plugin/README.md [hunk 5]
```diff
@@ -5,23 +5,38 @@ Contents
...
 
-This Intel SGX device plugin provides support for Intel SGX TEE under Kubernetes.
+- 3rd Generation Intel® Xeon® Scalable Platform, code-named “Ice Lake”
+- Intel® Xeon® E3
+- Intel® NUC Kit NUC7CJYH
+
+The SGX solution comes in three parts:
+
+- the [device plugin](/README.md#sgx-device-plugin)
+- the [admissing webhook](/README.md#sgx-admission-webhook)
```
Comment (intelkevinputnam) [inline cmd/sgx_plugin/README.md:36]: "Admission webhook" also the link should be: /README.md#admission-webhook

File: cmd/sgx_plugin/README.md [hunk 6]
```diff
@@ -37,17 +52,64 @@ the complete list of logging related options.
 
 ## Installation
 
-The below sections cover how to obtain, build and install this component.
+The below sections cover how to obtain, build and install the necessary Kubernetes SGX specific
```
Comment (intelkevinputnam) [inline cmd/sgx_plugin/README.md:55]: The "following" sections ...

File: cmd/sgx_plugin/README.md [hunk 7]
```diff
@@ -37,17 +52,64 @@ the complete list of logging related options.
...
+The SGX device plugin requires Linux Kernel SGX drivers to be available. These drivers
+are currently available via [RFC patches on Linux Kernel Mailing List](https://patchwork.kernel.org/project/intel-sgx/list/).
+RFC *v39* was used to validate what is written in this document.
+
+The hardware platform must support SGX Flexible Launch Control.
+
+### Pre-built images
+
+[Pre-built images](https://hub.docker.com/u/intel/)
+are available on the Docker hub. These images are automatically built and uploaded
```
Comment (intelkevinputnam) [inline cmd/sgx_plugin/README.md:74]: should be "Docker Hub" not "the Docker hub." Same with line 77 and 84.

File: cmd/sgx_plugin/README.md [hunk 8]
```diff
@@ -37,17 +52,64 @@ the complete list of logging related options.
...
 ### Prerequisites
 
 The component has the same basic dependancies as the
 [generic plugin framework dependencies](../../README.md#about).
 
-The SGX plugin requires Linux Kernel SGX drivers to be available. These drivers
-are currently available via RFC patches on Linux Kernel Mailing List.
+The SGX device plugin requires Linux Kernel SGX drivers to be available. These drivers
+are currently available via [RFC patches on Linux Kernel Mailing List](https://patchwork.kernel.org/project/intel-sgx/list/).
+RFC *v39* was used to validate what is written in this document.
```
[the remark this answers is not in the document]
Comment (mythi) [reply cmd/sgx_plugin/README.md:67]: the v39 tag works nice here, thanks!

File: cmd/sgx_plugin/README.md (hunk 1 above)
Comment (mythi) [reply cmd/sgx_plugin/README.md:238]: `sgx-aesmd-demo` is now in master.

**Comment by mythi** [comment]
@dcmiddle @intelkevinputnam thanks! I have your feedback addressed.

**Review by dcmiddle — DISMISSED** [review]

**Comment by mythi** [comment]
@ipuustin @rojkov ready for merging if you're OK

File: deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/add_hostnetwork.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: intelsgx-demo-job
+spec:
+  template:
+    spec:
+      hostNetwork: true
+      containers:
+        -
```
Comment (ipuustin) [inline deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/add_hostnetwork.yaml:10]: Is this line intentional, or should the `-` be on the line below`?

File: deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/add_sgx_default_qcnl_conf.yaml [hunk 10]
```diff
@@ -0,0 +1,22 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: intelsgx-demo-job
+spec:
+  template:
+    spec:
+      containers:
+        -
```
Comment (ipuustin) [inline deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/add_sgx_default_qcnl_conf.yaml:9]: Same here...

File: deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/kustomization.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+bases:
+  - ../sgx_ecdsa_aesmd_quote
+commonAnnotations:
+  sgx.intel.com/quote-provider: "intelsgx-demo-job-1"
+configMapGenerator:
+- name: sgx-attestation-conf
+  files:
+  - sgx_default_qcnl.conf
+patchesStrategicMerge:
+  - add_hostnetwork.yaml
```
Comment (ipuustin) [inline deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/kustomization.yaml:13]: indentation `<<`

**Review by ipuustin — COMMENTED** [review]
Looks good, just small stylistic nits!

File: deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/kustomization.yaml (hunk 11 above)
Comment (mythi) [reply deployments/sgx_enclave_apps/overlays/sgx_ecdsa_inproc_quote/kustomization.yaml]: fixed!

**Review by ipuustin — APPROVED** [review]
LGTM

**Review by rojkov — APPROVED** [review]

**Pull request merged**